embassy-boot/boot/src/firmware_updater.rs

```diff
@@ -182,6 +176,22 @@ impl FirmwareUpdater {
         self.set_magic(_aligned, SWAP_MAGIC, _state_and_dfu_flash).await
     }
 
+    /// Iterate through the DFU and process all bytes with the provided closure.
+    pub async fn incremental_hash<F: AsyncNorFlash>(
+        &mut self,
+        dfu_flash: &mut F,
+        update_len: u32,
+        aligned: &mut [u8],
+        mut update: impl FnMut(&[u8]),
+    ) -> Result<(), FirmwareUpdaterError> {
+        for offset in (0..update_len).step_by(aligned.len()) {
+            self.dfu.read(dfu_flash, offset, aligned).await?;
+            let len = core::cmp::min((update_len - offset) as usize, aligned.len());
+            update(&aligned[..len]);
```

Comment:
I originally wanted to use [Rust Crypto's `Digest` trait](https://github.com/RustCrypto/traits/tree/master/digest) but didn't as the ed25519 libs I chose didn't support it directly. However, those libs will support the trait in the future, and may even now do so. I think that, instead of this `update` closure, we should require something that implements the `Digest` trait instead, and provide our own implementations for the libs we are presently using. That shouldn't be hard to do.